testlapack/dgebal.go

```diff
@@ -0,0 +1,168 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"testing"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+	"github.com/gonum/lapack"
+)
+
+type Dgebaler interface {
+	Dgebal(job lapack.Job, n int, a []float64, lda int, scale []float64) (int, int)
+}
+
+func DgebalTest(t *testing.T, impl Dgebaler) {
+	rnd := rand.New(rand.NewSource(1))
+
+	for _, job := range []lapack.Job{lapack.None, lapack.Permute, lapack.Scale, lapack.PermuteScale} {
+		for _, n := range []int{0, 1, 2, 3, 4, 5, 6, 10, 18, 31, 53, 100} {
+			p := math.Ceil(0.02*float64(n)) / float64(n)
```

Comment:
Calling it `pNonzero`, it's the probability of an element being nonzero. Also added a check for n == 0 and n == 1 (pNonzero = 0.5).